File: cs/HAC.Tests.Unit/AbsoluteValuesSumMinimization.Test.cs
using NUnit.Framework;

namespace HAC.Tests.Unit
{
    public class AbsoluteValuesSumMinimizationTest
    {
        [TestCase(new[] { 2, 4, 7 }, 4)]
        [TestCase(new[] { 2, 4, 7, 6 }, 4)]
        [TestCase(new[] { 2, 4, 7, 6, 6 }, 7)]
        [TestCase(new[] { 2, 4, 7, 6, 6, 8 }, 7)]
        public void ShouldCalc(int[] data, int expected)
        {
            Assert.AreEqual(expected, HAC.AbsoluteValuesSumMinimization(data));
        }
    }
}

File: php/tests/AbsoluteValuesSumMinimizationTest.php
<?php

use PHPUnit\Framework\TestCase;
require "src/absolute_values_sum_minimization.php";

final class AbsoluteValuesSumMinimizationTest extends TestCase
{
    public function testAbsoluteValuesSumMinimization()
    {
        $data = [2, 4, 7];
        $actual = absolute_values_sum_minimization($data);
        $expected = 4;
        $this->assertEquals($expected, $actual);
  }
}
